### ISSUE TITLE
Orchestrator is retrying only once if activity function fails 

### ISSUE THREAD

**Issue by atear10** [Azure/azure-functions-durable-extension#1699]
### Description

I have configured retry mechanism in Orchestrator using built in retry method provided in Orchestrator , but if Activity function fails then orchestrator retries only once and then stops. below is the way I am setting up retry and calling

### Expected behavior

on fail it should retry 3 times on 60, 120, 240 intervals.

### Actual behavior

Retries only once after 60s.

### Relevant source code snippets
```csharp
// insert code snippet here
public MessageOrchestrator(IOptions<AppSettings> settings, ILoggerService logger)
        {
            _settings = settings.Value;
            _logger = logger;

            _retryOptions = new RetryOptions(
                                 firstRetryInterval: 60,
                                 maxNumberOfAttempts: 3
            {
                BackoffCoefficient = 2
            }
       }

await context.CallActivityWithRetryAsync<TransformedMessage>("MessageTransformer", _retryOptions, request)

### Known workarounds
> Provide a description of any known workarounds you used.

### App Details

- **Durable Functions extension version (e.g. v1.8.3)**:  2.2
- **Azure Functions runtime version (1.0 or 2.0)**:   3.0.3
- **Programming language used**:  C#

### Screenshots
> If applicable, add screenshots to help explain your problem.

### If deployed to Azure
> We have access to a lot of telemetry that can help with investigations. Please provide as much of the following information as you can to help us investigate!

- **Timeframe issue observed**:  
- **Function App name**:
- **Function name(s)**:
- **Azure region**:
- **Orchestration instance ID(s)**:  a8af1e3cd832484daeb199ec80cb164a
- **Azure storage account name**:  

> If you don't want to share your Function App or storage account name GitHub, please at least share the orchestration instance ID. Otherwise it's extremely difficult to look up information.

**Comment by ConnorMcMahon** [comment]
So I confirmed in the logs that only 1 retry was seen for instance id: a8af1e3cd832484daeb199ec80cb164a. I will look to see if this repros, but this is a scenario that is covered by our testing, so I'd be surprised if there was a regression here.

**Comment by YunchuWang** [comment]
I could not reproduce the retry stopping after the first retry on current `dev` (`22264d83f911c776d06363fdb79e0f864d27b40d`, extension source version 3.15.0).

I ran an in-process C# WebJobs host against the Azure Storage provider using a dedicated local Azurite instance. The repro used an instance constructor to initialize the retry options, a generic `CallActivityWithRetryAsync<string>` call, and an activity that throws on every invocation. The delays were real, not shortened or mocked:

```csharp
private readonly RetryOptions retryOptions =
    new RetryOptions(TimeSpan.FromSeconds(60), 3)
    {
        BackoffCoefficient = 2,
    };

public async Task<string> MessageOrchestrator(
    [OrchestrationTrigger] IDurableOrchestrationContext context)
{
    return await context.CallActivityWithRetryAsync<string>(
        nameof(MessageTransformer), retryOptions, context.GetInput<string>());
}

public string MessageTransformer([ActivityTrigger] string request)
{
    throw new InvalidOperationException("Intentional activity failure");
}
```

Persisted orchestration history showed the following in both configurations:

| Extended sessions | Activity failures | Retry delays | Final status |
| --- | --- | --- | --- |
| Disabled | 3 | 60s, 120s | Failed after the third attempt |
| Enabled | 3 | 60s, 120s | Failed after the third attempt |

`maxNumberOfAttempts: 3` includes the initial invocation, so it permits **two retries**, not three. For three retries with delays of 60, 120, and 240 seconds, set `maxNumberOfAttempts: 4`. These are delays between attempts, not absolute times since the initial call.

Closing as not reproducible on the current implementation. This does not invalidate the historical telemetry observation: the original extension 2.2 / Functions runtime 3.0.3 deployment was not reproduced here. If this still occurs on a supported version, please open a new issue with a complete runnable repro, exact versions and host configuration, and orchestration history showing where retries stop.
